routing/pathfind.go

```diff
@@ -179,7 +182,27 @@ func newRoute(amtToSend lnwire.MilliSatoshi, sourceVertex route.Vertex,
 			ChannelID:        edge.ChannelID,
 			AmtToForward:     amtToForward,
 			OutgoingTimeLock: outgoingTimeLock,
+			LegacyPayload:    true,
 		}
+
+		// We start out above by assuming that this node needs the
+		// legacy payload, as if we don't have the full
+		// NodeAnnouncement information for this node, then we can't
+		// assume it knows the latest features. If we do have a feature
+		// vector for this node, then we'll update the info now.
+		if edge.Node.Features != nil {
+			features := edge.Node.Features
+			currentHop.LegacyPayload = !features.HasFeature(
+				lnwire.TLVOnionPayloadOptional,
+			)
+		}
+
+		// If this is the last hop, then we'll populate any TLV records
+		// destined for it.
+		if i == len(pathEdges)-1 && len(finalDestRecords) != 0 {
+			currentHop.TLVRecords = finalDestRecords
```

Comment:
If the destination doesn't have the feature bit set, then we'll fail early in the path finding process. 

That information has already been threaded through in the form of the `finalDestRecords` field (in this isolated case). If it's populated, then we need the destination to support it, otherwise it doesn't matter. If you're referring to examining the proposed BOLT 11 invoice tags, then we can add more context here at a later time as that isn't yet fully flshed out.

Reply:
Not sure what you're referring to, this is where we fail earlier in path finding (before `findPath`) is even called: https://github.com/lightningnetwork/lnd/blob/10a44b35d2eb73606ba326af9f6a31d51ec51327/routing/pathfind.go#L366

Reply:
Yes, I know where it is failed earlier. My point is that if something should never happen, we should return an error rather than silently ignoring. The same amount of code, the same effect now, but an extra guard against future modifications that may violate this assumption that features will always be available.